SECURITY.md

```diff
@@ -2,58 +2,41 @@
 
 ## Supported Versions
 
-Vyper is currently in limited beta.
-This means that we only support the latest release and that you may encounter issues using it.
-It is un-audited software, use with caution.
+- each Vyper version in the range `<v0.1.0-beta.1, 0.3.10>` contains a high severity vulnerability
+  - if developing with such versions, please read https://github.com/vyperlang/vyper/security/advisories to learn how to work around the vulnerabilities
+- as of May 2024, the `0.4.0` release is the most secure and the most comprehensively reviewed one and is recommended for use in production environments
+- if a compiler vulnerability is found, a new compiler version with a patch will be released. The vulnerable version itself is not updated.
```

Comment:
We should clarify that older `0.x.y < 0.z.y` versions, where `x < z`, are however not patched. What I mean is that, let's say hypothetically, that there is a critical bug in the storage allocator of version `0.3.10` that also affects `0.4.0`. This means, that there will be a `0.4.1` patch release but _not_ a `0.3.11` patch as well. It's important to highlight this logic IMO.